resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -101,7 +101,23 @@ public ResourceInvoker match(HttpRequest request, int start)
                }
                else
                {
-                  String substring = path.substring(0, length);
+                  // must find the end of the matched pattern
+                  // and get the substring from 1st char thru end
+                  // of matched chars
+                  Pattern p = expression.getPattern();
+                  Matcher m = p.matcher(path);
+                  m.region(start, path.length());
+                  String substring = path;
+                  while(m.find()) {

```

Comment:
@liweinan thanks.
As for what @ronsigal is saying... I don't know, perhaps Rebecca can/want to add something. What I can say is that if we really expect multiple passes in the while loop, we should not keep on re-assigning the 'substring' variable to new String objects, we should instead store the latest indx value and eventually compute the substring using it at the end of the loop. Anyway, this can possibly be optimized later.